lib/slack/web/faraday/response/raise_error.rb

```diff
@@ -10,7 +10,10 @@ def on_complete(env)
 
             error_message =
               body['error'] || body['errors'].map { |message| message['error'] }.join(',')
-            raise Slack::Web::Api::Errors::SlackError.new(error_message, env.response)
+
+            error_class = Slack::Web::Api::Errors::ERROR_CLASSES[error_message]
```

Comment:
My original plan was to use `constantize` to turn the error message into the corresponding error class, but then I came across [this (old) post](http://gavinmiller.io/2016/the-safesty-way-to-constantize/) which claims `constantize` can be a security risk, so I took its advice and built a hash to map from the message to the class.

Reply:
This is a response from the server, so you can theoretically `.constantize`, but this will not be future proof. You'd have to capture handle new codes with a `rescue`. It's much better to use a map and not to rely on the exception for flow.

Reply:
Yes, that was another reason I didn't use `constantize`, because I would need to `rescue` for the fallback.